tiledb/sm/subarray/subarray.h

```diff
@@ -250,6 +250,13 @@ class Subarray {
   /** equivalent for older Query::set_subarray(const void *subarray); */
   Status set_subarray(const void* subarray);
 
+  /**
+   * Equivalent to older Query::set_subarray(const void *subarray).
```

Comment:
We should remove the reference to "older Query::set_subarray"... I don't think anyone reading this comment will know what that means.